src/statd/statd.c

```diff
@@ -221,56 +263,46 @@ static void sr_event_cb(struct ev_loop *, struct ev_io *w, int)
 	sr_subscription_process_events(sub->sr_sub, NULL, NULL);
 }
 
-static int sub_to_iface(struct statd *statd, const char *ifname)
+static int subscribe(struct statd *statd, char *model, char *xpath, const char *name,
+                     int (*cb)(sr_session_ctx_t *session, uint32_t, const char *,
+                               const char *, const char *, uint32_t,
+                               struct lyd_node **parent, void *priv))
 {
-	char path[XPATH_MAX] = {};
 	struct sub *sub;
 	int sr_ev_pipe;
-	int err;
-
-	/**
-	 * Skip internal interfaces (such as dsa0)
-	 *
-	 * NOTE: this is a good solution but it might be to slow if a lot of
-	 * new interfaces pops up at the same time. We don't want to delay
-	 * processing the netlink messages to much.
-	 */
-	if (ip_link_check_group(ifname, "internal") == 1)
-		return SR_ERR_OK;
+	sr_error_t err;
 
-	sub = sub_find_iface(&statd->subs, ifname);
+	sub = sub_find(&statd->subs, name);
```

Comment:
What will happen if someone names there interface "routes"?

Reply:
Good point, unlikely. But i fix it.